src/Adapter/Cart/QueryHandler/GetCartForViewingHandler.php

```diff
@@ -35,31 +35,48 @@
 use Group;
 use Order;
 use PrestaShop\PrestaShop\Adapter\ImageManager;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
 use PrestaShop\PrestaShop\Core\Domain\Cart\Exception\CartNotFoundException;
 use PrestaShop\PrestaShop\Core\Domain\Cart\Query\GetCartForViewing;
 use PrestaShop\PrestaShop\Core\Domain\Cart\QueryHandler\GetCartForViewingHandlerInterface;
 use PrestaShop\PrestaShop\Core\Domain\Cart\QueryResult\CartView;
+use PrestaShop\PrestaShop\Core\Localization\Locale;
+use PrestaShop\PrestaShop\Core\Localization\Locale\Repository as LocaleRepository;
 use Product;
 use StockAvailable;
-use Tools;
 use Validate;
 
 /**
  * @internal
  */
 final class GetCartForViewingHandler implements GetCartForViewingHandlerInterface
 {
+    /**
+     * @var LegacyContext
+     */
+    private $context;
+
     /**
      * @var ImageManager
      */
     private $imageManager;
 
+    /**
+     * @var Locale
+     */
+    private $locale;
+
     /**
      * @param ImageManager $imageManager
+     * @param LocaleRepository $repository
      */
-    public function __construct(ImageManager $imageManager)
+    public function __construct(ImageManager $imageManager, LocaleRepository $repository)
     {
         $this->imageManager = $imageManager;
+        $this->context = new LegacyContext();
```

Comment:
I think you'd better inject the locale via `'@=service("prestashop.adapter.legacy.context").getContext().language.getLocale()'` this avoid cerating a dependency with LegacyContext in this class (better to have it in the service definition)

Reply:
:heavy_check_mark: